src/dash-table/components/Button/ExportButton/index.tsx

```diff
@@ -0,0 +1,46 @@
+import XLSX from 'xlsx';
+import React, {
+    Component
+} from 'react';
+import { IDerivedData, IColumn, IVisibleColumn } from 'dash-table/components/Table/props';
+import { createWorkbook, createHeadings, createWorksheet } from './utils';
+import getHeaderRows from 'dash-table/derived/header/headerRows';
+
+interface IExportButtonProps {
+    export_format: string;
+    virtual_data: IDerivedData;
+    columns: IColumn[] & IVisibleColumn[];
+    export_header: string;
+}
+
+export default class ExportButton extends Component<IExportButtonProps, any> {
+
+    handleExport = () => {
+        const { columns, export_format, virtual_data, export_header } = this.props;
+        const columnID = columns.map(column => column.id);
+        const columnHeaders = columns.map(column => column.name);
+        const maxLength = getHeaderRows(columns);
+        const Heading = (export_header !== 'none') ? createHeadings(columnHeaders, maxLength) : [];
+        const ws = createWorksheet(Heading, virtual_data.data, columnID, export_header);
+        const wb = createWorkbook(ws, Heading, export_header);
+        if (export_format === 'xlsx') {
+            XLSX.writeFile(wb, 'Data.xlsx', {bookType: 'xlsx', type: 'buffer'});
+        } else if (export_format === 'csv') {
+            XLSX.writeFile(wb, 'Data.csv', {bookType: 'csv', type: 'buffer'});
+        }
+
+    }
```

Comment:
Feels like the component and utils both do half the job of seeing through the abstraction of the table export into the worksheet.

Reply:
this is because unit testing wouldn't accept export function from React Component